app/src/lib/stores/helpers/infer-comparison-branch.ts

```diff
@@ -0,0 +1,143 @@
+  branches: ReadonlyArray<Branch>,
+  repository: Repository,
+  currentPullRequest: PullRequest | null,
+  currentBranch: Branch | null,
+  getAheadBehind: (
+    repository: Repository,
+    range: string
+  ) => Promise<IAheadBehind | null>
+): Promise<Branch | null> {
+  if (currentPullRequest !== null) {
+    return _getTargetBranchOfPullRequest(branches, currentPullRequest)
+  }
+
+  const ghRepo = repository.gitHubRepository
+  if (ghRepo !== null) {
+    return ghRepo.fork === true && currentBranch !== null
+      ? _getDefaultBranchOfFork(
+          branches,
+          repository,
+          ghRepo,
+          currentBranch,
+          getAheadBehind
+        )
+      : _getDefaultBranchOfGithubRepo(branches, ghRepo)
+  }
+
+  return _getMasterBranch(branches)
+}
+
+/**
+ * For `inferComparisonBranch` case where inferring for local branches
+ *
+ * Returns `master`, the default branch of a Git repository
+ *
+ * @param branches The list of all branches for the repository
+ */
+function _getMasterBranch(branches: ReadonlyArray<Branch>): Branch | null {
+  return branches.find(b => b.name === 'master') || null
+}
+
+/**
+ * For `inferComparisonBranch` case where inferring for a repository
+ * hosted on GitHub, but with no forks
+ *
+ * Returns the default branch of the GitHub repository
+ *
+ * @param branches The list of all branches for the repository
+ * @param ghRepository The repository the branch belongs to
+ */
+function _getDefaultBranchOfGithubRepo(
+  branches: ReadonlyArray<Branch>,
+  ghRepository: GitHubRepository
+): Branch | null {
+  return branches.find(b => b.name === ghRepository.defaultBranch) || null
+}
+
+/**
+ * For `inferComparisonBranch` case where inferring  for a pull request
+ *
+ * Returns the base branch of the given pull request.
+ *
+ * @param branches The list of all branches for the repository
+ * @param pr The pull request to use for finding the branch
+ */
+function _getTargetBranchOfPullRequest(
+  branches: ReadonlyArray<Branch>,
+  pr: PullRequest
+): Branch | null {
+  return branches.find(b => b.name === pr.base.ref) || null
+}
+
+/**
+ * For `inferComparisonBranch` case where inferring for a forked repository
+ *
+ * Returns the default branch of the fork if it's ahead of `currentBranch`.
+ * Otherwise, the default branch of the parent is returned.
+ *
+ * @param branches The list of all branches for the repository
+ * @param repository The repository the branch belongs to
+ * @param ghRepository The github specific information for the repository
+ * @param currentBranch The branch we want the parent of
+ */
+async function _getDefaultBranchOfFork(
+  branches: ReadonlyArray<Branch>,
+  repository: Repository,
+  ghRepository: GitHubRepository,
+  currentBranch: Branch,
+  getAheadBehind: (
+    repository: Repository,
+    range: string
+  ) => Promise<IAheadBehind | null>
+): Promise<Branch | null> {
+  const defaultBranchName = ghRepository.defaultBranch
+
+  if (defaultBranchName === null) {
```

Comment:
it looks like the method `_getDefaultBranchOfGithubRepo` can handle the case where `ghRepository.defaultBranch` is null so I think we can skip to L121

Reply:
Good catch ⚡️